tests/test_multiplexer.py

```diff
@@ -75,14 +75,25 @@ def test_connect_twice():
     multiplexer.disconnect()
 
 
+def test_disconnect_twice():
+    """Test that connecting twice the multiplexer behaves correctly."""
+    multiplexer = Multiplexer([_make_dummy_connection()])
+
+    assert not multiplexer.connection_status.is_connected
+    multiplexer.connect()
+    assert multiplexer.connection_status.is_connected
+    multiplexer.disconnect()
+    multiplexer.disconnect()
```

Comment:
What are we testing here exactly? Maybe it can be clarified by adding 
``` python
assert not multiplexer.connection_status.is_connected
```
below first disconnect and then some logger output assertion on the second disconnect